src/Windowing/Silk.NET.Windowing.Common/Internals/ViewImplementationBase.cs

```diff
@@ -176,10 +179,13 @@ public void DoRender()
         public void DoUpdate()
         {
             // Check elapsed time
-            if ((Stopwatch.GetTimestamp() - _updateTimestamp) >= _updatePeriod)
+            var newTimeStamp = Stopwatch.GetTimestamp();
+            if ((newTimeStamp - _updateTimestamp) >= _updatePeriod)
             {
                 // Re-calculte the elapsed time, resetting the current timestamp
-                Update?.Invoke((-Interlocked.Exchange(ref _updateTimestamp, Stopwatch.GetTimestamp()) + _updateTimestamp) / Frequency);
+                var delta = (newTimeStamp - _updateTimestamp) / Frequency;
+                _updateTimestamp = newTimeStamp;
+                Update?.Invoke(delta);
```

Comment:
Update may have to be interlocked because update (once upon a time could be, we intend readd it) ran on another thread, it's only render that doesn't have to be interlocked.

Reply:
`Update?.Invoke((-Interlocked.Exchange(ref _updateTimestamp, Stopwatch.GetTimestamp()) + _updateTimestamp) / Frequency);` isn't thread-safe anyway, as it accesses `_updateTimestamp` twice non-atomically.  Making this code thread-safe is actually quite complex as there is a check before change.  I'll submit as a separate commit to show a thread-safe approach.